hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/IncrementalRelation.scala

```diff
@@ -220,15 +209,12 @@ class IncrementalRelation(val sqlContext: SQLContext,
 
       val sOpts = optParams.filter(p => !p._1.equalsIgnoreCase("path"))
 
-      val startInstantTime = optParams(DataSourceReadOptions.BEGIN_INSTANTTIME.key)
+      val startInstantTime = queryContext.getStartInstant.get()
       val startInstantArchived = commitTimeline.isBeforeTimelineStarts(startInstantTime)
-      val endInstantTime = optParams.getOrElse(DataSourceReadOptions.END_INSTANTTIME.key(), lastInstant.getTimestamp)
+      val endInstantTime = queryContext.getEndInstant.get()
```

Comment:
`startInstantArchived` and `endInstantArchived` can be avoided as `queryContext` already indicates whether there are instants in the archived timeline to pull.

Reply:
This is a good point, I think it also makes no sense to check if `endInstantArchived` is true here because 
1. end instant cannot come before start instant and 
2. We should scan the full table if either start or end instant is archived.

Will fix this